### ISSUE TITLE
Role and RoleBinding Kinds are not monitored properly by Spinnaker

### ISSUE THREAD

**Issue by sergedonaldp** [spinnaker/spinnaker#3451]
### Issue Summary:

When the role and rolebinding objects are deployed in Spinnaker, they fail to stabilize. So the deployment process will wait indefinitely and timeout. 

### Environment:
I am running spinnaker on  a GKE cluster to deploy a set of yaml manifests to another GKE cluster (GKE version used 1.10.7-gke.1).

### Feature Area: 

Deploy Manifest (stage)

### Description:

During a deployment of a role and rolebinding objects on a kubernetes cluster, Spinnaker waits indefinitely for these objects to stabilize and it leads to the deployment pipeline timing out. This causes our entire pipeline to fail, even though the role and rolebinding objects required are created by the pipeline. 
The expected behavior is for spinnaker to detect that the objects were created and mark them as stable then the pipeline can move to the next stages. 
![image-2018-10-05-16-58-05-748](https://user-images.githubusercontent.com/10876571/46956853-4fdd2480-d064-11e8-86af-58712a602612.png)

### Steps to Reproduce:
Minimal steps to reproduce the issue:
1. Create a delivery pipeline in spinnaker
2. Create a deploy with manifest stage to deploy a role manifest
3. Create a deploy with manifest stage to deploy a rolebinding manifest
4. Execute the pipeline

### Additional Details:

Logs from echo:

```
> 2018-10-05 15:24:13.050  WARN 1 --- [    handlers-20] c.n.s.o.c.t.m.WaitForManifestStableTask  : [localhost] Unable to read manifest 'roleBinding c4portal-chart' in 'test' for account myhome-k8s
> retrofit.RetrofitError: 404 
>     at retrofit.RetrofitError.httpError(RetrofitError.java:40)
>     at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:388)
>     at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240)
>     at com.sun.proxy.$Proxy97.getManifest(Unknown Source)
>     at com.netflix.spinnaker.orca.clouddriver.DelegatingOortService.getManifest(DelegatingOortService.java:48)
>     at com.netflix.spinnaker.orca.clouddriver.tasks.manifest.WaitForManifestStableTask.execute(WaitForManifestStableTask.java:83)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1$1.invoke(RunTaskHandler.kt:90)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1$1.invoke(RunTaskHandler.kt:54)
>     at com.netflix.spinnaker.orca.q.handler.AuthenticationAware$sam$java_util_concurrent_Callable$0.call(AuthenticationAware.kt)
>     at com.netflix.spinnaker.security.AuthenticatedRequest.lambda$propagate$1(AuthenticatedRequest.java:79)
>     at com.netflix.spinnaker.orca.q.handler.AuthenticationAware$DefaultImpls.withAuth(AuthenticationAware.kt:49)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withAuth(RunTaskHandler.kt:54)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1.invoke(RunTaskHandler.kt:89)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1.invoke(RunTaskHandler.kt:54)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$withTask$1.invoke(RunTaskHandler.kt:163)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$withTask$1.invoke(RunTaskHandler.kt:54)
>     at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withTask$1.invoke(OrcaMessageHandler.kt:48)
>     at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withTask$1.invoke(OrcaMessageHandler.kt:32)
>     at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withStage$1.invoke(OrcaMessageHandler.kt:58)
>     at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withStage$1.invoke(OrcaMessageHandler.kt:32)
>     at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.withExecution(OrcaMessageHandler.kt:67)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withExecution(RunTaskHandler.kt:54)
>     at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.withStage(OrcaMessageHandler.kt:54)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withStage(RunTaskHandler.kt:54)
>     at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.withTask(OrcaMessageHandler.kt:41)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withTask(RunTaskHandler.kt:54)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withTask(RunTaskHandler.kt:156)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.handle(RunTaskHandler.kt:66)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.handle(RunTaskHandler.kt:54)
>     at com.netflix.spinnaker.q.MessageHandler$DefaultImpls.invoke(MessageHandler.kt:36)
>     at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.invoke(OrcaMessageHandler.kt)
>     at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.invoke(RunTaskHandler.kt:54)
>     at com.netflix.spinnaker.orca.q.audit.ExecutionTrackingMessageHandlerPostProcessor$ExecutionTrackingMessageHandlerProxy.invoke(ExecutionTrackingMessageHandlerPostProcessor.kt:47)
>     at com.netflix.spinnaker.q.QueueProcessor$pollOnce$1$1.run(QueueProcessor.kt:83)
>     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
>     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
>     at java.lang.Thread.run(Thread.java:748)
```

**Comment by lwander** [comment]
Did you add a namespace to the cluster role or cluster role binding objects? While the root cause is a bug fixed in the upcoming release, you can't add namespaces to cluster-scoped objects since this breaks how the lookup happens.

**Comment by sergedonaldp** [comment]
Hi @lwander  we tried without a namespace since they were cluster-scoped objects.

**Comment by daviddyball** [comment]
I'm seeing this issue as well. My deployment has a `ServiceAccount`, `ClusterRole`, `ClusterRoleBinding`, `Role` and `RoleBinding` and it's breaking on the `Role`/`RoleBinding`. There _is_ `metadata.namespace` set on the `Role` and `RoleBinding` as they should indeed be namespaced, but the `ClusterRole` and `ClusterRoleBinding` objects don't have `metadata.namespace` set on them.

**Comment by lwander** [comment]
This was fixed here: https://github.com/spinnaker/clouddriver/pull/3052, will be available in 1.10.1 as well
